ui/app/src/modules/Preview/Tools/AudiencesPanel.tsx

```diff
@@ -15,24 +15,250 @@
 import ToolPanel from './ToolPanel';
-import { defineMessages } from 'react-intl';
+import { defineMessages, FormattedMessage, useIntl } from 'react-intl';
+import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
+import Divider from '@material-ui/core/Divider';
+import Grid from '@material-ui/core/Grid';
+import FormHelperText from '@material-ui/core/FormHelperText';
+import Button from '@material-ui/core/Button';
+import { useSelection, useStateResource } from '../../../utils/hooks';
+import { ErrorBoundary } from '../../../components/ErrorBoundary';
+import LoadingState from '../../../components/SystemStatus/LoadingState';
+import { useDispatch } from 'react-redux';
+import { setActiveModel, updateAudiencesPanelModel } from '../../../state/actions/preview';
+import { ContentTypeField } from '../../../models/ContentType';
+import { nnou, nou, reversePluckProps } from '../../../utils/object';
+import GlobalState from '../../../models/GlobalState';
+import ContentInstance from '../../../models/ContentInstance';
+import Input from '../../../components/Controls/FormEngine/Input';
+import Dropdown from '../../../components/Controls/FormEngine/Dropdown';
+import CheckboxGroup from '../../../components/Controls/FormEngine/CheckboxGroup';
+import DateTime from '../../../components/Controls/FormEngine/DateTime';
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    fab: {
+      margin: theme.spacing(1),
+    },
+    formControl: {
+      width: '100%',
+      '& .MuiFormGroup-root': {
+        marginLeft: '10px',
+      },
+      '& .MuiInputBase-root': {
+        marginTop: '12px !important',
+      }
+    },
+    ActionButton: {
+      padding: '6px',
+      marginLeft: '5px'
+    },
+    InputLabel: {
+      position: 'relative'
+    },
+    PanelMargin: {
+      margin: '0 15px',
+    },
+    textField: {
+      width: '100%',
+    },
+    actionBTN:{
+      margin:'18px 0 80px 15px',
+      '& .MuiButton-contained':{
+        marginRight: '8px',
+      }
+    },
+    divider: {
+      margin: '23px 0 10px',
+    }
+  }),
+);
 
 const translations = defineMessages({
   audiencesPanel: {
     id: 'craftercms.ice.audiences.title',
     defaultMessage: 'Audience Targeting'
+  },
+  audiencesPanelLoading: {
+    id: 'craftercms.ice.audiences.loading',
+    defaultMessage: 'Retrieving targeting options'
   }
 });
 
-export default function AudiencesPanel() {
+const controlsMap = {
+  'dropdown': Dropdown,
+  'checkbox-group': CheckboxGroup,
+  'date-time': DateTime,
+  'input': Input
+};
+
+interface AudiencesPanelUIProps {
+  audiencesResource: any;
+  model: ContentInstance;
+  modelApplying: boolean;
+  modelApplied: boolean;
+  onFormChange: Function;
+  onSaveModel: Function;
+  onSetDefaults: Function;
+}
+
+export function AudiencesPanelUI(props: AudiencesPanelUIProps) {
+
+  const classes = useStyles({});
+  const {
+    audiencesResource,
+    model,
+    modelApplying,
+    onFormChange,
```

Comment:
Do a `onFieldChange(fieldId: string, value: any)` internally. It would spread update the model prop and then call on~Form~Change prop with the whole model updated for the container to dispatch. 

We can now remove onSetDefaults: the set defaults button would take the contentType defaults, create a model and call onChange.